tests/corpus/parsing/test_cha.py

```diff
@@ -1,46 +1,38 @@
-import os
-import tempfile
 import pytest
-import requests
 from sktalk.corpus.conversation import Conversation
 from sktalk.corpus.parsing.cha import ChaFile
 
 
-class TestChaFile:
-    urls = [
-        "https://ca.talkbank.org/data-orig/GCSAusE/01.cha",
-        "https://ca.talkbank.org/data-orig/GCSAusE/02.cha",
-        "https://ca.talkbank.org/data-orig/GCSAusE/03.cha"
-    ]
-
-    @pytest.fixture(params=urls)
-    def download_file(self, request):
-        remote = request.param
-        response = requests.get(remote, timeout=5)
-        response.raise_for_status()
+@pytest.fixture
+def path_source():
+    return "tests/testdata/file01.cha"
 
-        ext = os.path.splitext(remote)[1]
 
-        with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as temp_file:
-            temp_file.write(response.content)
-            temp_file_path = temp_file.name
+@pytest.fixture
+def cha_info():
+    n_utterances = 13
+    participants = {'A', 'B'}
+    languages = ['eng']
+    return n_utterances, participants, languages
 
-        yield temp_file_path
 
-        os.remove(temp_file_path)
-
-    @pytest.mark.parametrize("download_file", urls, indirect=True)
-    def test_parse(self, download_file):
-        parsed_cha = ChaFile(download_file).parse()
+class TestChaFile:
+    def test_parse(self, path_source, cha_info):
+        n_utterances, participants, languages = cha_info
+        cha_utts, cha_meta = ChaFile(path_source).parse()
+        assert cha_meta["source"] == path_source
+        assert {u.participant for u in cha_utts} == participants
+        assert cha_meta["Languages"] == languages
+        assert len(cha_utts) == n_utterances
```

Comment:
It seems more intuitive to me if you compare the resulting `ChaFile` to the expected `ChaFile()`.
```suggestion
        assert cha_meta == {"source":  path_source, "Languages": languages}
        assert cha_utterances = expected_utterances # TODO: fill in `expected_utterances`
```
Just to compare all values in the resulting objects, preventing unexpected changes in any of their properties.

Reply:
Ive applied this to the metadata, but it would be quite complex to do this to the utterances I think... and it wouldn't add much, yet could create failing tests with any changes in the Utterance class itself.